File: Day111_2025/Q111_GFG.java
class Q111_GFG {
    int missingNum(int[] arr) {
        int n = arr.length + 1; // since one number is missing
        int xorSum = 0;
        
        for (int i = 1; i <= n; i++) {
            xorSum ^= i;
        }
        
        for (int num : arr) {
            xorSum ^= num;
        }
        
        return xorSum;
    }
}

File: Day111_2025/Q111_GFG.py
class Solution:
    def missingNum(self, arr):
        n = len(arr) + 1
        xor_sum = 0
        
        for i in range(1, n + 1):
            xor_sum ^= i
            
        for num in arr:
            xor_sum ^= num
            
        return xor_sum
